api4/api.go

```diff
@@ -145,12 +145,18 @@ type API struct {
 	srv        *app.Server
 	schema     *graphql.Schema
 	BaseRoutes *Routes
+
+	// The following are used for testing purposes only
+	knownAPIsByScope map[model.Scope][]string
```

Comment:
@agnivade these are filled when routes are initialized (`handlers.go`), and verified in `scope_test.go` against hardcoded expected values. When a new API is added or an existing one is modified, the expected values will have to be adjusted, and it would be the decision point to keep it as "internal", or make it available to OAuth2 apps by assigning it a scope (including a `*:*` to make it unrestricted)

Reply:
Thanks, took another look and all good now. I was thinking that we were just matching the map keys one by one, and if a new handler gets added it will get missed out. Now I see we are appending the keys to a slice and comparing the entire slice.